examples/core-triangle/App.zig

```diff
@@ -24,43 +25,49 @@ pub fn init(
     core.on_tick = app_mod.id.tick;
     core.on_exit = app_mod.id.deinit;
 
-    // Create our shader module
-    const shader_module = core.device.createShaderModuleWGSL("shader.wgsl", @embedFile("shader.wgsl"));
-    defer shader_module.release();
-
-    // Blend state describes how rendered colors get blended
-    const blend = gpu.BlendState{};
-
-    // Color target describes e.g. the pixel format of the window we are rendering to.
-    const color_target = gpu.ColorTargetState{
-        .format = core.windows.get(core.main_window, .framebuffer_format).?,
-        .blend = &blend,
-    };
-
-    // Fragment state describes which shader and entrypoint to use for rendering fragments.
-    const fragment = gpu.FragmentState.init(.{
-        .module = shader_module,
-        .entry_point = "frag_main",
-        .targets = &.{color_target},
-    });
-
-    // Create our render pipeline that will ultimately get pixels onto the screen.
-    const label = @tagName(mach_module) ++ ".init";
-    const pipeline_descriptor = gpu.RenderPipeline.Descriptor{
-        .label = label,
-        .fragment = &fragment,
-        .vertex = gpu.VertexState{
-            .module = shader_module,
-            .entry_point = "vertex_main",
-        },
-    };
-    const pipeline = core.device.createRenderPipeline(&pipeline_descriptor);
-
-    // Store our render pipeline in our module's state, so we can access it later on.
-    app.* = .{
-        .title_timer = try mach.time.Timer.start(),
-        .pipeline = pipeline,
-    };
+    if (core.windows.getAll(core.main_window)) |cw| {
+        const core_window = cw;
+        if (core_window.device) |device| {
```

Comment:
presumably both these if statements, if the logic inside of them doesn't run, the app wouldn't work. So I think we should replace them with non-null assertions:

```
const core_window = core.windows.getAll(core.main_window).?;
const device = core_window.device.?;
```